src/Catalog/Icons/IconProcessor.cs

```diff
@@ -59,22 +60,28 @@ public class IconProcessor : IIconProcessor
             return destinationUri;
         }
 
-        public async Task DeleteIcon(
-            Storage destinationStorage,
+        public async Task DeleteIconAsync(
+            IStorage destinationStorage,
             string destinationStoragePath,
             CancellationToken cancellationToken,
             string packageId,
             string normalizedPackageVersion)
         {
             _logger.LogInformation("Deleting icon blob {IconPath}", destinationStoragePath);
-            if (destinationStorage.Exists(destinationStoragePath))
+            var iconUri = new Uri(destinationStorage.BaseAddress, destinationStoragePath);
+            try
+            {
+                await destinationStorage.DeleteAsync(iconUri, cancellationToken, new DeleteRequestOptionsWithAccessCondition(AccessCondition.GenerateIfExistsCondition()));
```

Comment:
When would the icon not exist? What do you expect to happen here if the icon does not exist? An exception to be thrown? The no-op to occur? IIRC if you specify `If-Match: *` and no blob is there a precondition failure exception will be thrown. Might want to confirm.

Reply:
This will happen, for example, when we see the first package details leaf for some package identity with icon url specified and we fail to retrieve that URL. Service does not track which package exist or not, so we need to clean up unconditionally in this case.

If icon does not exist, the call no-ops, I had it tested.